pulp_plugin_template/tests/functional/utils.py

```diff
@@ -31,67 +29,63 @@ def set_up_module():
     require_pulp_plugins({'pulp_plugin_template'}, SkipTest)
 
 
-def gen_plugin_template_remote(**kwargs):
+def gen_plugin_template_remote(url=None, **kwargs):
     """Return a semi-random dict for use in creating a plugin_template Remote.
 
     :param url: The URL of an external content source.
     """
-    remote = gen_remote(PLUGIN_TEMPLATE_FIXTURE_URL)
+    if url is None:
```

Comment:
Just set ```url=PLUGIN_TEMPLATE_FIXTURE_URL``` in the signature.  The ```is None``` check is good practice with mutable data structures like lists, but strings are immutable so it's not necessary.

Reply:
After seeing this in pulp_file i was unsure whether strings are immutable. Thanks for clearing this up.

Reply:
Np!  I just double checked

```
>>> def test_list(url=[]):
...     url += 'a'
...     print(url)
... 
>>> test_list()
['a']
>>> test_list()
['a', 'a']
>>> test_list()
['a', 'a', 'a']
>>> def test_string(url=""):
...     url += 'a'
...     print(url)
... 
>>> test_string()
a
>>> test_string()
a
>>> test_string()
a

```